extensions/2.0/Vendor/EXT_feature_metadata/schema/class.property.schema.json

```
@@ -17,26 +17,20 @@
     "type": {
       "type": "string",
       "enum": [
-        "INT8",
-        "UINT8",
-        "INT16",
-        "UINT16",
-        "INT32",
-        "UINT32",
-        "INT64",
-        "UINT64",
-        "FLOAT32",
-        "FLOAT64",
-        "BOOLEAN",
-        "STRING",
-        "ENUM",
+        "SINGLE",
+        "VEC2",
+        "VEC3",
+        "VEC4",
+        "MAT2",
+        "MAT3",
+        "MAT4",
         "ARRAY"
       ],
-      "description": "The property type. If `ENUM` is used, then `enumType` must also be specified. If `ARRAY` is used, then `componentType` must also be specified. `ARRAY` is a fixed-length array when `componentCount` is defined, and variable-length otherwise."
+      "description": "Element type represented by each property value. `ARRAY` is a fixed-length array when `componentCount` is defined and variable-length otherwise. Types `VECN` and `MATN` have implicit component counts of `N` and `N ⨉ N` respectively."
```

I know Oxford commas are debatable in academic writing. Though I wonder, does the standards community have an opinion on it? I'd rather have the comma if it makes things less ambiguous.

```suggestion
      "description": "Element type represented by each property value. `ARRAY` is a fixed-length array when `componentCount` is defined, and variable-length otherwise. Types `VECN` and `MATN` have implicit component counts of `N` and `N ⨉ N` respectively."
```

---

I would replace the last sentence with 3 for clarity. the key 3 points are:

1. `VECN` correspond to vectors with `N` components
2. `MATN` corresponds to an `N x N` matrix
3. `VECN` and `MATN` ignore the component count (as that's only for `ARRAY`

This way we explicitly mention "vector" and "matrix" in the text here.

---

I don't think this is technically an Oxford comma , but I agree it sounds weird as written — I restructured this a bit.

Pushed a change that I think covers the other points, please take a look!